app/models/manageiq/providers/elasticsearch/datawarehouse_manager.rb

```diff
@@ -0,0 +1,95 @@
+module ManageIQ::Providers
+  class Elasticsearch::DatawarehouseManager < ManageIQ::Providers::DatawarehouseManager
+    require 'elasticsearch'
+
+    require_nested :RefreshParser
+    require_nested :RefreshWorker
+    require_nested :Refresher
+
+    include AuthenticationMixin
```

Comment:
Did you try adding a provider from the API?
I think it will fail unless you override `required_credential_fields`.
See https://github.com/ManageIQ/manageiq/pull/13877